### PR TITLE
Performance and cleanup

### PR DESCRIPTION
**Pull request by rilango**

### CODE REVIEW

File: benchmark/cuchembm/__main__.py [hunk 1]
```diff
@@ -81,6 +78,54 @@ def save_metric_results(mode_name, metric_list, output_dir, return_predictions):
     metric_df.to_csv(csv_file_path, index=False, mode='a', header=write_header)
 
 
+def identity_dataset(cfg, inferrer):
```
Comment (mlgill) [inline benchmark/cuchembm/__main__.py:81]: Does this function generate the parameters for the datasets? If so, maybe it should be named to something like `create_dataset`? Identity has a mathematical meaning that might be confusing?

**Review by mlgill — APPROVED** [review]
@rilango Approved, minor changes requested, but this is all up to your discretion.

File: setup/docker_compose.yml [hunk 2]
```diff
@@ -65,4 +61,17 @@ services:
         reservations:
           devices:
           - driver: nvidia
-            capabilities: [gpu]
\ No newline at end of file
+            capabilities: [gpu]
+            count: 1
+
+  nginx:
```
[the remark this answers is not in the document]
Comment (rilango) [reply setup/docker_compose.yml:67]: For load-balancing incoming gRPC requests.

One of the changes is meant to start multiple instances of MegaMolBART instances and process multiple sampling request. NGINX will distribute the incoming requests among MegaMolBART instances.

File: benchmark/cuchembm/data/training.py [hunk 3]
```diff
@@ -1,60 +1,60 @@
-import os
-import sqlite3
-import logging
-from typing import List
+# import os
```
[the remark this answers is not in the document]
Comment (rilango) [reply benchmark/cuchembm/data/training.py:1]: Removed.

**Review by mlgill — APPROVED** [review]

**Pull request merged**